src/CFamily/CMake/VsDevCmdEnvironmentProvider.cs

```diff
@@ -0,0 +1,231 @@
+
+        private const int SCRIPT_TIMEOUT_MS = 4000;
+
+        private readonly IVsInfoService vsInfoService;
+        private readonly ILogger logger;
+
+        // Interfaces for testing
+        private readonly IProcessFactory processFactory;
+        private readonly IFileSystem fileSystem;
+
+
+        public VsDevCmdEnvironmentProvider(IVsInfoService vsInfoService, ILogger logger)
+            : this(vsInfoService, logger, new ProcessFactory(), new FileSystem())
+        {
+            this.vsInfoService = vsInfoService;
+            this.logger = logger;
+        }
+
+        internal /* for testing */ VsDevCmdEnvironmentProvider(IVsInfoService vsInfoService, ILogger logger,
+            IProcessFactory processFactory, IFileSystem fileSystem)
+        {
+            this.vsInfoService = vsInfoService;
+            this.logger = logger;
+            this.processFactory = processFactory;
+            this.fileSystem = fileSystem;
+        }
+
+        public IReadOnlyDictionary<string, string> Get(string scriptParams)
+        {
+            try
+            {
+                string filePath = Path.Combine(vsInfoService.InstallRootDir, RelativePathToBatchFile);
+                LogDebug($"VsDevCmd location: {filePath}");
+
+                if (!fileSystem.File.Exists(filePath))
+                {
+                    logger.WriteLine(Resources.VsDevCmd_FileNotFound, filePath);
+                    return null;
+                }
+
+                var capturedOutput = ExecuteVsDevCmd(filePath, scriptParams);
+                var settings = ParseOutput(capturedOutput);
+
+                if (settings == null || settings.Count == 0)
+                {
+                    logger.WriteLine(Resources.VsDevCmd_NoSettingsFound);
+                    return null;
+                }
+
+                LogDebug($"Settings fetched successfully. Count: {settings.Count}");
+                return settings;
+            }
+            catch (Exception ex) when (!ErrorHandler.IsCriticalException(ex))
+            {
+                logger.WriteLine(Resources.VsDevCmd_ErrorFetchingSettings, ex);
+            }
+
+            return null;
+        }
+
+        internal /* for testing */ string UniqueId { get; private set;}
+
+        private IList<string> ExecuteVsDevCmd(string batchFilePath, string scriptParams)
+        {
+            UniqueId = Guid.NewGuid().ToString();
+            var beginToken = "SONARLINT_BEGIN_CAPTURE " + UniqueId;
+            var endToken = "SONARLINT_END_CAPTURE " + UniqueId;
+
+            ProcessStartInfo startInfo = new ProcessStartInfo()
+            {
+                FileName = Environment.GetEnvironmentVariable("COMSPEC"),
+                Arguments = "/U " +                             // Unicode input / output
+                "/K " +                                         // Execute the command and remain active. Need to do this or we can't fetch the env vars 
```

Comment:
`/K` means the process won't terminate, which is why there is code that explicitly kills it, either in the event of a timeout, or after the "end capture" marker has been read.